server/pxf-service/src/main/java/org/greenplum/pxf/service/security/BaseSecurityService.java

```
@@ -55,58 +54,52 @@ public BaseSecurityService(SecureLogin secureLogin, UGIProvider ugiProvider) {
+                LOG.debug("Creating remote user = {}", remoteUser);
                 userGroupInformation = ugiProvider.createRemoteUser(remoteUser, loginUser, isSecurityEnabled);
             }
 
-            LOG.debug("{} Retrieved proxy user {} for server {}", contextId, userGroupInformation, serverName);
-            LOG.debug("Performing request for gpdb_user = {} as [remote_user = {} service_user = {} login_user ={}] with{} impersonation",
+            LOG.debug("Retrieved proxy user {} for server {}", userGroupInformation, serverName);
+            LOG.debug("Performing request for gpdb_user = {} as [remote_user={} service_user={} login_user={}] with{} impersonation",
                     gpdbUser, remoteUser, serviceUser, loginUser.getUserName(), isUserImpersonation ? "" : "out");
             // Execute the servlet chain as that user
             return userGroupInformation.doAs(action);
-        } catch (UndeclaredThrowableException ute) {
-            exceptionDetected = true;
-            // unwrap the real exception thrown by the action
-            throw new IOException(ute.getCause());
-        } catch (InterruptedException ie) {
+        } catch (Exception e) {
             exceptionDetected = true;
-            throw new IOException(ie);
+            throw e;
         } finally {
-            LOG.debug("{} Releasing UGI resources. {}",
-                    contextId, exceptionDetected ? " Exception while processing" : "");
+            LOG.debug("Releasing UGI resources. {}", exceptionDetected ? " Exception while processing" : "");
             try {
                 if (userGroupInformation != null) {
                     ugiProvider.destroy(userGroupInformation);
                 }
             } catch (Throwable t) {
-                LOG.error("{} Error releasing UGI resources", contextId, t);
+                LOG.warn("Error releasing UGI resources, ignored.", t);
```

why do we go from logged error to ignored error listed as warn?

---

failure during destroying UGI when the request is complete does not impact the outcome seen by user, so this is not an error (smith that prevented successful outcome), but rather a warning about meeting an unexpected condition. We did not throw it before either, just changing the log level here to `warn` and stating that it is ignored by the logic flow.